accounts/store_test.go

```diff
@@ -37,9 +36,43 @@ func TestAccountStore(t *testing.T) {
 	_, err = store.NewAccount(ctx, 123, time.Time{}, "0011223344556677")
 	require.ErrorContains(t, err, "is not allowed as it can be mistaken")
 
+	now := time.Now()
+
 	// Update all values of the account that we can modify.
+	//
+	// Update the balance and expiry.
+	err = store.UpdateAccountBalanceAndExpiry(
+		ctx, acct1.ID, fn.Some(int64(-500)), fn.Some(now),
+	)
+	require.NoError(t, err)
+
+	// Add 2 payments.
+	_, err = store.UpsertAccountPayment(
+		ctx, acct1.ID, lntypes.Hash{12, 34, 56, 78}, 123456,
+		lnrpc.Payment_FAILED,
+	)
+	require.NoError(t, err)
+
+	_, err = store.UpsertAccountPayment(
+		ctx, acct1.ID, lntypes.Hash{34, 56, 78, 90}, 789456123789,
+		lnrpc.Payment_SUCCEEDED,
+	)
+	require.NoError(t, err)
+
+	// Add 2 invoices.
+	err = store.AddAccountInvoice(
+		ctx, acct1.ID, lntypes.Hash{12, 34, 56, 78},
+	)
+	require.NoError(t, err)
+	err = store.AddAccountInvoice(
+		ctx, acct1.ID, lntypes.Hash{34, 56, 78, 90},
+	)
+	require.NoError(t, err)
```

Comment:
nit: would personally put these after updating the in-memory `acct1`, as that makes more logical sense, and would also then change it to something like:

```
	for paymentHash, entry := range acct1.Payments {
		_, err = store.UpsertAccountPayment(
			ctx, acct1.ID, paymentHash, entry.FullAmount,
			entry.Status,
		)
		require.NoError(t, err)
	}

	for paymentHash := range acct1.Invoices {
		err = store.AddAccountInvoice(ctx, acct1.ID, paymentHash)
		require.NoError(t, err)
	}
```

To not have to repeat hashes and amounts.

Reply:
> To not have to repeat hashes and amounts.

I kind of like keeping tests very explicit. Else a bug in one version can be transferred to the other version.

For example, if you are testing an `func Add(x, y int) int` function, then one should test explicitly like:

```
x := 1
y := 2
expected := 3
result := Add(x, y)
require.Equal(3, result)
```
 
instead of:

```
x := 1
y := 2
expected := x+y
result := Add(x, y)
require.Equal(expected, result)
```